test/tests/materials/derivative_material_interface/ad_material_chaining.i

```diff
@@ -0,0 +1,152 @@
+# material properties within DerivativeParsedMaterials
+#
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 5
+  ny = 5
+  xmin = 0
+  xmax = 1
+  ymin = 0
+  ymax = 1
+[]
+
+[Variables]
+  [./eta1]
+  [../]
+  [./eta2]
+  [../]
+[]
+
+[BCs]
+  [./left]
+    variable = eta1
+    boundary = left
+    type = DirichletBC
+    value = 0
+  [../]
+  [./right]
+    variable = eta1
+    boundary = right
+    type = DirichletBC
+    value = 1
+  [../]
+  [./top]
+    variable = eta2
+    boundary = top
+    type = DirichletBC
+    value = 0
+  [../]
+  [./bottom]
+    variable = eta2
+    boundary = bottom
+    type = DirichletBC
+    value = 1
+  [../]
+[]
+
+[Materials]
+  # T1 := (eta1+1)^4
+  [./term]
+    type = ADDerivativeParsedMaterial
+    f_name= T1
+    args = 'eta1'
+    function = '(eta1+1)^4'
+    derivative_order = 4
+  [../]
+
+  # in this material we substitute T1 explicitly
+  [./full]
+    type = ADDerivativeParsedMaterial
+    args = 'eta1 eta2'
+    f_name = F1
+    function = '(1-eta2)^4+(eta1+1)^4'
+  [../]
+  # in this material we utilize the T1 derivative material property
+  [./subs]
+    type = ADDerivativeParsedMaterial
+    args = 'eta1 eta2'
+    f_name = F2
+    function = '(1-eta2)^4+T1'
+    material_property_names = 'T1(eta1)'
+  [../]
+
+  # calculate differences between the explicit and indirect substitution version
+  # the use if the T1 property should include dT1/deta1 contributions!
+  # This also demonstrated the explicit use of material property derivatives using
+  # the D[...] syntax.
+  [./diff0]
+    type = ADParsedMaterial
+    f_name = D0
+    function = '(F1-F2)^2'
+    material_property_names = 'F1 F2'
+  [../]
+  [./diff1]
+    type = ADParsedMaterial
+    f_name = D1
+    function = '(dF1-dF2)^2'
+    material_property_names = 'dF1:=D[F1,eta1] dF2:=D[F2,eta1]'
+  [../]
+  [./diff2]
+    type = ADParsedMaterial
+    f_name = D2
+    function = '(d2F1-d2F2)^2'
+    material_property_names = 'd2F1:=D[F1,eta1,eta1] d2F2:=D[F2,eta1,eta1]'
+  [../]
+
+  # check that explicitly pulling a derivative yields the correct result by
+  # taking the difference of the manually calculated 1st derivative of T1 and the
+  # automatic derivative dT1 pulled in through dT1:=D[T1,eta1]
+  [./diff3]
+    type = ADParsedMaterial
+    f_name = E0
+    function = '(dTd1-(4*(eta1+1)^3))^2'
+    args = eta1
+    material_property_names = 'dTd1:=D[T1,eta1]'
+  [../]
+[]
+
+[Kernels]
+  [./eta1diff]
+    type = Diffusion
+    variable = eta1
+  [../]
+  [./eta2diff]
+    type = Diffusion
+    variable = eta2
+  [../]
+[]
+
+[Postprocessors]
+  [./D0]
+    type = ADElementIntegralMaterialProperty
+    mat_prop = D0
+  [../]
+  [./D1]
+    type = ADElementIntegralMaterialProperty
+    mat_prop = D1
+  [../]
+  [./D2]
+    type = ADElementIntegralMaterialProperty
+    mat_prop = D2
+  [../]
+  [./E0]
+    type = ADElementIntegralMaterialProperty
+    mat_prop = E0
+  [../]
+[]
```

Comment:
@dschwen the gold file for these just has all zeros. Is that really what it should be?